### ISSUE TITLE
Issues when trying to build from git

### ISSUE THREAD

**Issue by Ketrel** [linuxdeploy/linuxdeploy#254]
I'm trying to build from git (17ca786 specifically)

When I do, using cmake in a _build directory, I get an error.  To test further, I tried using the ci/build.sh script (arch x86_64, tried use_static both ways) and I ultimately get the build failing out with the same error

```
[ 28%] Building CXX object src/subprocess/CMakeFiles/linuxdeploy_subprocess.dir/subprocess.cpp.o
/home/builder/dev/git/linuxdeploy/src/subprocess/subprocess.cpp: In member function ‘linuxdeploy::subprocess::subprocess_result linuxdeploy::subprocess::subprocess::run() const’:
/home/builder/dev/git/linuxdeploy/src/subprocess/subprocess.cpp:49:38: error: variable ‘std::array<linuxdeploy::subprocess::subprocess::run() const::PipeState, 2> buffers’ has initializer but incomplete type
   49 |             std::array<PipeState, 2> buffers = { 
      |                                      ^~~~~~~
/home/builder/dev/git/linuxdeploy/src/subprocess/subprocess.cpp:98:85: error: no matching function for call to ‘linuxdeploy::subprocess::subprocess_result::subprocess_result(<brace-enclosed initializer list>)’
   98 |             return subprocess_result{exit_code, buffers[0].buffer, buffers[1].buffer};
      |                                                                                     ^
In file included from /home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess.h:11,
                 from /home/builder/dev/git/linuxdeploy/src/subprocess/subprocess.cpp:11:
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:23:13: note: candidate: ‘linuxdeploy::subprocess::subprocess_result::subprocess_result(int, linuxdeploy::subprocess::subprocess_result_buffer_t, linuxde
   23 |             subprocess_result(int exit_code, subprocess_result_buffer_t stdout_contents,
      |             ^~~~~~~~~~~~~~~~~
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:23:13: note:   conversion of argument 2 would be ill-formed:
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:16:15: note: candidate: ‘linuxdeploy::subprocess::subprocess_result::subprocess_result(const linuxdeploy::subprocess::subprocess_result&)’
   16 |         class subprocess_result {
      |               ^~~~~~~~~~~~~~~~~
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:16:15: note:   candidate expects 1 argument, 3 provided
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:16:15: note: candidate: ‘linuxdeploy::subprocess::subprocess_result::subprocess_result(linuxdeploy::subprocess::subprocess_result&&)’
/home/builder/dev/git/linuxdeploy/include/linuxdeploy/subprocess/subprocess_result.h:16:15: note:   candidate expects 1 argument, 3 provided
```

gcc version 13.2.1 20230801 (GCC)  
cmake version 3.27.1

**Comment by gsohler** [comment]
got same issue. any advice ?

**Comment by petrmanek** [comment]
Just tried to build linuxdeploy myself, got to the same place.

Happens both with gcc and clang:

 * clang version 16.0.6, Target: x86_64-pc-linux-gnu
 * g++ (GCC) 13.2.1 20230801
 * cmake version 3.27.6

**Comment by vorporeal** [comment]
Ran into this too; looks like two files are using `std::array` but missing `#include <array>`.  Will put together a PR with a fix, but until I do, it's easy to fix locally.

**Comment by vorporeal** [comment]
Opened https://github.com/linuxdeploy/linuxdeploy/pull/257 with a fix.

**Comment by TheAssassin** [comment]
Fixed by #257.
